fe/fe-core/src/main/java/org/apache/doris/analysis/AnalyzeStmt.java

```diff
@@ -17,19 +17,181 @@
+        return this.tableIdToColumnName;
+    }
+
+    public Map<String, String> getProperties() {
+        return this.properties;
+    }
 
+    @Override
+    public void analyze(Analyzer analyzer) throws UserException {
+        super.analyze(analyzer);
+
+        // step1: analyze database and table
+        if (this.dbTableName != null) {
+            String dbName;
+            if (Strings.isNullOrEmpty(this.dbTableName.getDb())) {
+                dbName = analyzer.getDefaultDb();
+            } else {
+                dbName = ClusterNamespace.getFullName(analyzer.getClusterName(), this.dbTableName.getDb());
+            }
+
+            // check db
+            if (Strings.isNullOrEmpty(dbName)) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+            this.db = analyzer.getCatalog().getDbOrAnalysisException(dbName);
+
+            // check table
+            String tblName = this.dbTableName.getTbl();
+            if (Strings.isNullOrEmpty(tblName)) {
+                this.tables = this.db.getTables();
+                for (Table table : this.tables) {
+                    checkAnalyzePriv(dbName, table.getName());
+                }
+            } else {
+                Table table = this.db.getTableOrAnalysisException(tblName);
+                this.tables = Collections.singletonList(table);
+                checkAnalyzePriv(dbName, table.getName());
+            }
+
+            // check column
+            if (this.columnNames == null || this.columnNames.isEmpty()) {
+                setTableIdToColumnName();
+            } else {
+                Table table = this.db.getTableOrAnalysisException(tblName);
+                for (String columnName : this.columnNames) {
+                    Column column = table.getColumn(columnName);
+                    if (column == null) {
+                        ErrorReport.reportAnalysisException(ErrorCode.ERR_WRONG_COLUMN_NAME, columnName);
+                    }
+                }
+                this.tableIdToColumnName.put(table.getId(), this.columnNames);
+            }
+        } else {
+            // analyze the current default db
+            String dbName = analyzer.getDefaultDb();
+            if (Strings.isNullOrEmpty(dbName)) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+            this.db = analyzer.getCatalog().getDbOrAnalysisException(dbName);
+            this.tables = this.db.getTables();
+            for (Table table : this.tables) {
+                checkAnalyzePriv(dbName, table.getName());
+            }
+            setTableIdToColumnName();
+        }
+
+        // step2: analyze properties
+        if (this.properties != null) {
+            for (Map.Entry<String, String> pros : this.properties.entrySet()) {
+                if (!"cbo_statistics_task_timeout_sec".equals(pros.getKey())) {
```

Comment:
Please declare acceptable property set
For code examples see 'CreateRoutineLoadStmt'